### PR TITLE
[Media Common] Set Scanout by default

### PR DESCRIPTION
**Pull request by vcheah**
recommend flagging all potential scanout surfaces as SCANOUT

### CODE REVIEW

**Review by XinfengZhang — APPROVED** [review]
LGTM, suppose only ddi allocate surface is using "Media" words? + @Jexu @Effiey to review

File: media_softlet/linux/common/os/xe/mos_bufmgr_xe.c [hunk 1]
```diff
@@ -1295,6 +1295,9 @@ mos_bo_alloc_xe(struct mos_bufmgr *bufmgr,
      */
     create.cpu_caching = alloc->ext.cpu_cacheable ? DRM_XE_GEM_CPU_CACHING_WB : DRM_XE_GEM_CPU_CACHING_WC;
 
+    if (strcmp(alloc->name, "MEDIA") == 0 || strcmp(alloc->name, "Media") == 0)
+        create.flags |= DRM_XE_GEM_CREATE_FLAG_SCANOUT;
```
Comment (Jexu) [inline media_softlet/linux/common/os/xe/mos_bufmgr_xe.c:1299]: Cpu caching will become WC inside KMD when setting scanout, but umd/gmm may still record this surface with cpu cachable somewhere and want WB caching. Will have any conflict between umd/gmm and kmd?

Comment (vcheah) [reply media_softlet/linux/common/os/xe/mos_bufmgr_xe.c]: Added additional check to ensure that the caching is WC before we set the scanout flag.

**Review by Jexu — APPROVED** [review]

**Review by XinfengZhang — APPROVED** [review]

File: media_softlet/linux/common/os/xe/mos_bufmgr_xe.c [hunk 2]
```diff
@@ -1295,6 +1295,10 @@ mos_bo_alloc_xe(struct mos_bufmgr *bufmgr,
      */
     create.cpu_caching = alloc->ext.cpu_cacheable ? DRM_XE_GEM_CPU_CACHING_WB : DRM_XE_GEM_CPU_CACHING_WC;
 
+    if ((strcmp(alloc->name, "MEDIA") == 0 || strcmp(alloc->name, "Media") == 0)
+        && create.cpu_caching == DRM_XE_GEM_CPU_CACHING_WC)
+            create.flags |= DRM_XE_GEM_CREATE_FLAG_SCANOUT;
+
```
Comment (Effieyu0) [inline media_softlet/linux/common/os/xe/mos_bufmgr_xe.c:1301]: if set DRM_XE_GEM_CREATE_FLAG_SCANOUT would need 64K alignment, also need to align it in  UMD.

Comment (vcheah) [reply media_softlet/linux/common/os/xe/mos_bufmgr_xe.c:1301]: Hi @Effieyu0, I would like to check 64K alignment are you referring to https://patchwork.freedesktop.org/series/135900/ ?

Comment (vcheah) [reply media_softlet/linux/common/os/xe/mos_bufmgr_xe.c:1301]: I read the https://patchwork.freedesktop.org/series/135900/ and it seems the alignment are set within the kernel space. Please do let me know if i missed out anything or if i'm not looking at the correct kernel patch.

Comment (Effieyu0) [reply media_softlet/linux/common/os/xe/mos_bufmgr_xe.c:1301]: Yes. KMD will do 64K alignment.

Comment (vcheah) [reply media_softlet/linux/common/os/xe/mos_bufmgr_xe.c:1301]: Noted with thanks.  Please do let me know if there are any other changes that i need to do.

**Comment by vcheah** [comment]
I've just rebased.  @Effieyu0, could you please help to review?

**Review by Effieyu0 — APPROVED** [review]

**Pull request merged**